src/js/actions/libraries.js

```diff
@@ -0,0 +1,272 @@
+        }).then(function (path) {
+            var placeObj = libraryAdapter.placeElement(docRef, element, path, location);
+
+            return descriptor.playObject(placeObj);
+        });
+    };
+    createLayerFromElement.reads = [locks.JS_LIBRARIES, locks.JS_DOC];
+    createLayerFromElement.writes = [locks.JS_LIBRARIES, locks.JS_DOC];
+
+    /**
+     * Marks the given library ID as the active one
+     *
+     * @param {string} id
+     *
+     * @return {Promise}
+     */
+    var selectLibrary = function (id) {
+        return this.dispatchAsync(events.libraries.LIBRARY_SELECTED, { id: id });
+    };
+    selectLibrary.reads = [];
+    selectLibrary.writes = [locks.JS_LIBRARIES];
+
+    /**
+     * Creates a new library with the given name
+     *
+     * @param {string} name
+     * @return {Promise.<Library>} Resolves to the created library
+     */
+    var createLibrary = function (name) {
+        var libStore = this.flux.store("library"),
+            libraryCollection = libStore.getLibraryCollection(),
+            newLibrary = libraryCollection.createLibrary(name);
+
+        return this.dispatchAsync(events.libraries.LIBRARY_CREATED, { library: newLibrary })
+            .then(function () {
+                return newLibrary;
+            });
+    };
+    createLibrary.reads = [];
+    createLibrary.writes = [locks.CC_LIBRARIES, locks.JS_LIBRARIES];
+
+    /** 
+     * Removes the current library from the collection
+     *
+     * @return {Promise}
+     */
+    var removeCurrentLibrary = function () {
+        var libStore = this.flux.store("library"),
+            libraryCollection = libStore.getLibraryCollection(),
+            currentLibrary = libStore.getCurrentLibrary();
+
+        if (!libraryCollection || !currentLibrary) {
+            return Promise.resolve();
+        }
+
+        var payload = {
+            id: currentLibrary.id
+        };
+
+        return Promise.fromNode(function (cb) {
+            libraryCollection.removeLibrary(currentLibrary, cb);
+        }).bind(this).then(function () {
+            this.dispatchAsync(events.libraries.LIBRARY_REMOVED, payload);
+        });
+    };
+    removeCurrentLibrary.reads = [locks.CC_LIBRARIES, locks.JS_LIBRARIES];
+    removeCurrentLibrary.writes = [locks.CC_LIBRARIES, locks.JS_LIBRARIES];
+
+    var beforeStartup = function () {
+        var dependencies = {
+            // Photoshop on startup will grab the port of the CC Library process and expose it to us
+            vulcanCall: function (requestType, requestPayload, responseType, callback) {
+                descriptor.getProperty("application", "designSpaceLibrariesIMSInfo")
+                    .then(function (imsInfo) {
+                        var port = imsInfo.port;
+
+                        callback(JSON.stringify({ port: port }));
+                    });
+            }
+        };
+
+        // SHARED_LOCAL_STORAGE flag forces websocket use
+        CCLibraries.configure(dependencies, {
```

Comment:
Having looked at the source, it shuts down any existing interfaces if they're already up, so it's safe to call repeatedly.